Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -236,10 +250,8 @@ public async Task PutAsync(string remotename, string localname, CancellationToke
     /// <exception cref="Exception">Exceptions arising from either code execution</exception>
     public async Task PutAsync(string remotename, Stream input, CancellationToken cancellationToken)
     {
-        // If we don't have the folder ID cached, lets fetch it with the create folder api
-        if (_CachedFolderID == null)
-            await CreateFolderAsync(cancellationToken).ConfigureAwait(false);
-
+        _CachedFolderID ??= GetFolderId().Await();
```

Comment:
This should be `_CachedFolderID ??= await GetFolderId(cancellationToken).ConfigureAwait(false)`